### PR TITLE
Carousel throws JS error after first panel is added, required reload to work #1386

### LINKED ISSUE #1386: Carousel throws JS error after first panel is added, required reload to work
Reported by @msagolj in pre-release testing.

How to reproduce:

- create a page, open browser console
- drop an carousel
- drop a component on the carousel

=> you get the following error in browser console:

```
clientlib-base.js:1726 Uncaught TypeError: Cannot read property 'dataset' of undefined
    at getDataLayerId (clientlib-base.js:1726)
    at navigate (clientlib-base.js:1592)
    at Array.<anonymous> (clientlib-base.js:1236)
    at MessageChannel.receiveMessage (messaging.js:5)
```

- add more panels using the carousel edit dialog

- try to switch between the panels using the panel switcher in the carousel toolbar

=> each time the above error occurs , no switch happens

- Reload the page

Now the carousel works without errors
NOTE : Happens on 6.4.8.3, 6.5.7, cloud QS
NOTE : does not happen for the other 2 panel containers, accordion and tabs

Error occurs in this location it seems:
https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/clientlibs/site/js/carousel.js#L631

### PR DESCRIPTION
**Pull request by vladbailescu**
Fixes #1386

### CODE REVIEW

**Review by bartoszglow — APPROVED** [review]

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js [hunk 1]
```diff
@@ -533,7 +533,7 @@
      * @returns {String} dataLayerId or undefined
      */
     function getDataLayerId(item) {
-        if (item.dataset.cmpDataLayer) {
+        if (item && item.dataset.cmpDataLayer) {
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js:536]: to be on the safe side I'd add: 
`if (item && item.dataset && item.dataset.cmpDataLayer) {`

File: content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/clientlibs/site/js/carousel.js [hunk 2]
```diff
@@ -628,7 +628,7 @@
      * @returns {String} dataLayerId or undefined
      */
     function getDataLayerId(item) {
-        if (item.dataset.cmpDataLayer) {
+        if (item && item.dataset.cmpDataLayer) {
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/carousel/v1/carousel/clientlibs/site/js/carousel.js:631]: see above comment

File: content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/site/js/tabs.js [hunk 3]
```diff
@@ -357,7 +357,7 @@
      * @returns {String} dataLayerId or undefined
      */
     function getDataLayerId(item) {
-        if (item.dataset.cmpDataLayer) {
+        if (item && item.dataset.cmpDataLayer) {
```
Comment (jckautzmann) [inline content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/clientlibs/site/js/tabs.js:360]: see above comment

File: content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js (hunk 1 above)
Comment (vladbailescu) [reply content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js:536]: `dataset` is part of the `HTMLElement` [API](https://developer.cdn.mozilla.net/en-US/docs/Web/API/HTMLOrForeignElement/dataset) and will not be null/undefined.

**Pull request merged**